### PR TITLE
`azurerm_logic_app_standard` - update resource to use `go-azure-sdk` and add `public_network_access`

### LINKED ISSUE #23809: Support for publicNetworkAccess switch
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

azurerm_logic_app_standard does not have an argument to control **Public Network Access**. 
Can we add an argument like `public_network_access` which can be set to `true` or `false` to control the public access. 

### New or Affected Resource(s)/Data Source(s)

azurerm_logic_app_standard 

### Potential Terraform Configuration

```hcl
resource "azurerm_logic_app_standard" "example" {
  name                       = "test-azure-functions"
  location                   = azurerm_resource_group.example.location
  resource_group_name        = azurerm_resource_group.example.name
  app_service_plan_id        = azurerm_app_service_plan.example.id
  storage_account_name       = azurerm_storage_account.example.name
  storage_account_access_key = azurerm_storage_account.example.primary_access_key
  public_network_access      = "false"

  site_config {
    linux_fx_version = "DOCKER|mcr.microsoft.com/azure-functions/dotnet:3.0-appservice"
  }

  app_settings = {
    "DOCKER_REGISTRY_SERVER_URL"      = "https://<server-name>.azurecr.io"
    "DOCKER_REGISTRY_SERVER_USERNAME" = "username"
    "DOCKER_REGISTRY_SERVER_PASSWORD" = "password"
  }

}
```

### References

_No response_

### PR DESCRIPTION
**Pull request by stephybun**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

- Updates the Logic App Standard resource to use `hashicorp/go-azure-sdk`
- adds the `public_network_access` property on the site properties level in both the resource and the data source
- fixes the half baked documentation for resource and data source
- To handle the behaviour described below the `site_config.public_network_access_enabled` property has been deprecated and `public_network_access` will control setting this on the site properties as well as the site_config level of the resource

The `site_config.public_network_access_enabled` and `public_network_access` properties are intertwined
My observations are:
- setting `site_config.public_network_access_enabled` will also set `public_network_access`
- setting `public_network_access` will not set `site_config.public_network_access_enabled`
- `site_config.public_network_access_enabled` can be updated (as in API will return whatever value we send to it)
- `public_network_access` can not be updated

## Testing 

4.x mode:
image
5.0 mode:
image

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_logic_app_standard` - add support for the `public_network_access` property [GH-00000]

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #23809
Supersedes #24842

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

**Comment by premaarya** [comment]
We are waiting for this feature/bug fix, any idea how long we need to wait for it to available?

**Review by mbfrahry — APPROVED** [review]
LGTM

File: internal/services/logic/logic_app_standard_resource.go [hunk 1]
```diff
@@ -1024,32 +1056,38 @@ func schemaLogicAppStandardIpRestriction() *pluginsdk.Schema {
 	}
 }
 
-func flattenLogicAppStandardAppSettings(input map[string]*string) map[string]string {
+func flattenLogicAppStandardAppSettings(input *map[string]string) map[string]string {
```
Comment (jackofallops) [inline internal/services/logic/logic_app_standard_resource.go:1059]: as per the data source, can probably be removed and replaced with `pointer.From()`.

File: website/docs/r/logic_app_standard.html.markdown [hunk 2]
```diff
@@ -181,17 +185,19 @@ The `site_config` block supports the following:
 
 * `app_scale_limit` - (Optional) The number of workers this Logic App can scale out to. Only applicable to apps on the Consumption and Premium plan.
 
+* `auto_swap_slot_name` - The Auto-swap slot name.
```
Comment (jackofallops) [inline website/docs/r/logic_app_standard.html.markdown:188]:
```suggestion
* `auto_swap_slot_name` - (Optional) The Auto-swap slot name.
```

**Review by jackofallops — APPROVED** [review]
Thanks @stephybun - just a couple minor things to check, but otherwise LGTM 🦦

**Pull request merged**